scripts/bootstrap

```diff
@@ -0,0 +1,200 @@
+        install_packages "$@" || throw "Failed to install requested packages"
+    fi
+}
+
+entware_install() {
+    bootstrap_wget || throw "bootstrap_wget() failed!"
+    (wget -O- "https://raw.githubusercontent.com/$REMARKABLE_ENTWARE_REPO_AUTHOR/remarkable_entware/master/entware_install.sh" | sh) \
+        || throw "Failed to download and/or install remarkable_entware."
+    log INFO "Entware was installed successfully."
+}
+
+entware_reenable() {
+    bootstrap_wget || throw "bootstrap_wget() failed!"
+    (wget -O- "https://raw.githubusercontent.com/$REMARKABLE_ENTWARE_REPO_AUTHOR/remarkable_entware/master/entware_reenable.sh" | sh) \
+        || throw "Failed to download and/or install remarkable_entware."
+    log INFO "Entware was reenabled successfully."
+}
+
+ensure_opkg_available_from_bashrc() {
+    # Run by ensure_opkg_available
+    if ! grep '.*PATH=.*/opt/bin.*' /home/root/.bashrc | grep -v '^#' >/dev/null; then
+        # Path is not in .bashrc
+        echo -e '\n# Path added by bootstrap.sh' >> /home/root/.bashrc
+        echo 'PATH="/opt/bin:/opt/sbin:$PATH"' >> /home/root/.bashrc
+        log WARN '/opt/bin and /opt/sbin were not added to $PATH in ~/.bashrc. Fixed automatically.'
+    fi
+}
+
+ensure_opkg_available() {
+    if command -v opkg &> /dev/null; then # opkg command not found
+        export PATH="/opt/bin/:/opt/sbin/:$PATH"
+        messages="$(ensure_opkg_available_from_bashrc)"
+        if [[ -n $messages ]]; then
+            echo "$messages" # Print the caught output
+            log WARN "Please re-connect to your reMarkable after this" \
+                     "script is finished. Otherwise opkg and other" \
+                     "installed binaries won't be recognized!"
+        fi
+    else
+        ensure_opkg_available_from_bashrc
+    fi
+}
+
+ensure_toltec_repo_added() {
+    if ! grep "^src/gz\b.*\bhttps://toltec\.delab\.re/" /opt/etc/opkg.conf >/dev/null 2>&1; then
+        # No active toltec repo found in opkg.conf
+        echo "src/gz toltec https://toltec.delab.re/$TOLTEC_BRANCH" >> /opt/etc/opkg.conf
+        opkg update || throw "Failed to update opkg after adding toltec $TOLTEC_BRANCH repo"
+        log INFO "Added the toltec $TOLTEC_BRANCH repo to opkg"
+    fi
+}
+
+ensure_opt_wget_and_ca_certs_installed() {
+    # Those packages might miss if someone had already
+    # installed entware before.
+    packages=()
+    [ -d /opt/etc/ssl/certs ] || packages+=(ca-certificates)
+    [ -f /opt/bin/wget ] || packages+=(wget)
+
+    if [[ ${#packages[@]} -gt 0 ]]; then
+        opkg update && opkg install "${packages[@]}" || throw "Failed to install packages for better https support"
+        log INFO "Installed missing recommended packages for better https support"
+    fi
+}
+
+install_packages() {
+    [ $# -eq 0 ] && return
+
+    if [ $# -eq 1 ]; then
+        log INFO "Installing package: $1"
+    else
+        log INFO "Installing packages: $*"
+    fi
+
+    opkg install "$@"
+}
+
+log() {
```

Comment:
The `log()` and `throw()` functions in this script could be merged with the `status()`, `section()` and `error()` functions in `scripts/package-lib`.

Reply:
Good idea. Though we would then also supply those scripts as well. Do have an idea on how to use them without straight copying them?

Reply:
We could have a build step which simply `cat`s the lib and this script, then publishes the result to `toltec.delab.re`.

Reply:
Good idea. This would make the repo more complex though. Would be nice if you could add that as I'm not that comfortable with GH Actions, yet.

Reply:
Another drawback is that the resulting file's checksum will change anytime we change lib, and therefore we will need to remember to manually update the one-liner install command accordingly even if we only update the `package-lib`.

Reply:
Leaving this as is for now to move forward.